blockly-core/core/ui/block_space/block_space.js

```diff
@@ -815,6 +818,20 @@ Blockly.BlockSpace.prototype.getScrollableSize = function(metrics) {
 };
 
 /**
+ * Returns the maximum possible scrolling offsets (+x right, +y down) for
+ * this BlockSpace.
+ * @returns {{x: number, y: number}}
+ */
+Blockly.BlockSpace.prototype.getMaxScrollOffsets = function() {
+  var metrics = this.getMetrics();
+  var blockSpaceSize = this.getScrollableSize(metrics);
+  return {
+    x: -blockSpaceSize.width + metrics.viewWidth,
```

Comment:
What do you think of writing this as  "metrics.viewWidth - blockSpaceSize.Width".  That seems somewhat more intuitive and conventional and the reader is less likely to miss the leading unary minus.  (Likewise for the next line.)